.jenkins/pytorch/build-asan.sh

```diff
@@ -16,6 +16,6 @@ export ASAN_OPTIONS=detect_leaks=0:symbolize=1
 
 # TODO: Make the ASAN flags a more unified env var
 CC="clang" CXX="clang++" LDSHARED="clang --shared" \
-  CFLAGS="-fsanitize=address -shared-libasan" \
+  CFLAGS="-fsanitize=address -fsanitize=undefined -shared-libasan" \
```

Comment:
`-fsanitize=undefined` enables a large subset of the available UBSAN checks. There is also 
* `-fsanitize=nullability` to help detect incorrectly used nullability attributes, should be fairly easy to enable and fix. 
* `-fsanitize=unsigned-integer-overflow`, while not an undefined behavior it might be useful to turn on.

UBSAN by default prints only one line of log, you can make it print the whole stack trace if you define environment variable:
`UBSAN_OPTIONS=print_stacktrace=1`.
https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#available-checks